### PR TITLE
Fixed sync and pathing issues with wp-cli

### PR DESCRIPTION
**Pull request by EvanK**
Scope of this PR has changed significantly since it was started, so summarizing the changes:

* [x] [Defines a local wp path](https://github.com/evolution/wordpress/pull/64/files#diff-8b4375835c2a4e397b434c8db725312aR32) in bootstrap
* [x] [Reuses said local path](https://github.com/evolution/wordpress/pull/64/files#diff-ec349a38925b8585bf0ad61acbfc8a54R6) in generated local.rb
* [x] [Adds util task](https://github.com/evolution/wordpress/pull/64/files#diff-9df3b7e8ffe49ea41076385f663daa72R2) to calculate working wp path for given stage
* [x] Uses said util task anywhere that wp-cli commands are run by capistrano, [including the standalone wp task](https://github.com/evolution/wordpress/pull/64/files#diff-54aff2561818567205dd9b6a97a53e8bR5)
* [x] Prevents syncing up/down on a working stage of `local`

--

Fixed an issue where `db:down` was incorrectly using the remote stage's `wp_path` against the local target stage (and failing).

Also added exceptions when doing a down from local stage, ~~or an up from a remote stage~~ -- this is wrong...confused myself with my own terminology.

Conceptually speaking we have three distinct stages to worry about during sync:
* **working stage** - this is what the current capistrano command is running under; eg `cap {$stage} {$task}`
* **source stage** - this is the stage _from which_ we're syncing; in `down` this is the remote, and in `up` this is local
* **target stage** - this is the stage _to which_ we're syncing; in `down` this is local, and in `up` this is the remote

To review, the sync tasks should be preventing an up or down with a **working stage** of `local`.

--

Output from this issue (that I found entirely by accident, while testing #62):

```bash
$ bundle exec cap staging evolve:db:down
...some output removed for brevity sake...
INFO[af1302ab] Running /usr/bin/env vagrant ssh local -c 'cd /vagrant && mysql -uroot -D "example_db_local" < example_db.2015-06-25.135213.sql' on localhost
DEBUG[af1302ab] Command: /usr/bin/env vagrant ssh local -c 'cd /vagrant && mysql -uroot -D "example_db_local" < example_db.2015-06-25.135213.sql'
INFO[af1302ab] Finished in 2.580 seconds with exit status 0 (successful).
INFO[ae35d28b] Running /usr/bin/env vagrant ssh local -c 'cd /var/www/example.com/staging/feature-branch/current/web/wp && wp search-replace "://staging.example.com/" "://local.example.com/" --path="/var/www/example.com/staging/feature-branch/current/web/wp" --url="http://local.example.com/"' on localhost
DEBUG[ae35d28b] Command: /usr/bin/env vagrant ssh local -c 'cd /var/www/example.com/staging/feature-branch/current/web/wp && wp search-replace "://staging.example.com/" "://local.example.com/" --path="/var/www/example.com/staging/feature-branch/current/web/wp" --url="http://local.example.com/"'
DEBUG[89bdde17] Running /usr/bin/env [ -f /var/log/evolution/wordpress.log ] && [ $(stat -c %U /var/log/evolution/wordpress.log) == 'deploy' ] on staging.example.com
DEBUG[89bdde17] Command: [ -f /var/log/evolution/wordpress.log ] && [ $(stat -c %U /var/log/evolution/wordpress.log) == 'deploy' ]
DEBUG[89bdde17] Finished in 0.052 seconds with exit status 0 (successful).
INFO[10e7aaea] Running /usr/bin/env echo Thu\ Jun\ 25\ 13:52:24\ 2015\	ekaufman@discordia.local\	evolve:db:down\	Failure >> /var/log/evolution/wordpress.log on staging.example.com
DEBUG[10e7aaea] Command: /usr/bin/env echo Thu\ Jun\ 25\ 13:52:24\ 2015\	ekaufman@discordia.local\	evolve:db:down\	Failure >> /var/log/evolution/wordpress.log
INFO[10e7aaea] Finished in 0.050 seconds with exit status 0 (successful).
cap aborted!
SSHKit::Command::Failed: vagrant exit status: 256
vagrant stdout: Nothing written
vagrant stderr: bash: line 0: cd: /var/www/example.com/staging/feature-branch/current/web/wp: No such file or directory
/Users/ekaufman/GIT/CMN/Example.com/vendor/bundle/ruby/2.0.0/gems/sshkit-1.5.1/lib/sshkit/command.rb:97:in `exit_status='
/Users/ekaufman/GIT/CMN/Example.com/vendor/bundle/ruby/2.0.0/gems/sshkit-1.5.1/lib/sshkit/backends/local.rb:56:in `block in _execute'
/Users/ekaufman/GIT/CMN/Example.com/vendor/bundle/ruby/2.0.0/gems/sshkit-1.5.1/lib/sshkit/backends/local.rb:37:in `tap'
/Users/ekaufman/GIT/CMN/Example.com/vendor/bundle/ruby/2.0.0/gems/sshkit-1.5.1/lib/sshkit/backends/local.rb:37:in `_execute'
/Users/ekaufman/GIT/CMN/Example.com/vendor/bundle/ruby/2.0.0/gems/sshkit-1.5.1/lib/sshkit/backends/local.rb:26:in `execute'
/Users/ekaufman/GIT/CMN/Example.com/bower_components/evolution-wordpress/lib/capistrano/tasks/db.rake:65:in `block (4 levels) in <top (required)>'
/Users/ekaufman/GIT/CMN/Example.com/vendor/bundle/ruby/2.0.0/gems/sshkit-1.5.1/lib/sshkit/backends/local.rb:14:in `instance_exec'
/Users/ekaufman/GIT/CMN/Example.com/vendor/bundle/ruby/2.0.0/gems/sshkit-1.5.1/lib/sshkit/backends/local.rb:14:in `run'
/Users/ekaufman/GIT/CMN/Example.com/vendor/bundle/ruby/2.0.0/gems/sshkit-1.5.1/lib/sshkit/dsl.rb:10:in `run_locally'
/Users/ekaufman/GIT/CMN/Example.com/bower_components/evolution-wordpress/lib/capistrano/tasks/db.rake:61:in `block (3 levels) in <top (required)>'
/Users/ekaufman/GIT/CMN/Example.com/vendor/bundle/ruby/2.0.0/gems/capistrano-3.2.1/lib/capistrano/application.rb:15:in `run'
/Users/ekaufman/GIT/CMN/Example.com/vendor/bundle/ruby/2.0.0/gems/capistrano-3.2.1/bin/cap:3:in `<top (required)>'
/Users/ekaufman/GIT/CMN/Example.com/vendor/bundle/ruby/2.0.0/bin/cap:23:in `load'
/Users/ekaufman/GIT/CMN/Example.com/vendor/bundle/ruby/2.0.0/bin/cap:23:in `<main>'
Tasks: TOP => evolve:db:down
(See full trace by running task with --trace)
```

### CODE REVIEW

File: lib/yeoman/templates/lib/capistrano/deploy/local.rb [hunk 1]
```diff
@@ -3,4 +3,4 @@
   user:         fetch(:user),
   ssh_options:  fetch(:ssh_options)
 
-set :wp_path,   "/vagrant/web/wp"
```
Comment (EvanK) [inline lib/yeoman/templates/lib/capistrano/deploy/local.rb]: While this _does_ modify a generated file, it has no functional difference and should not constitute a bc break

**Comment by EvanK** [comment]
Added a [util task to calculate path for wp-cli commands](https://github.com/evolution/wordpress/blob/681a14dcb7d33c4125612985ea524d01a47cc7aa/lib/capistrano/tasks/util.rake#L2), taking into account that the working branch may not match what's deployed to the stage and substituting `master` as necessary.

**Comment by germanny** [comment]
Sorry for finding all the issues!!!

**Comment by EvanK** [comment]
Frankly, I'm surprised this one didn't bite us much sooner!

I appreciate the help flushing out edge cases

**Pull request merged**